src/sentry/static/sentry/app/views/organizationIntegrations/addIntegrationButton.jsx

```diff
@@ -90,8 +94,10 @@ export default class AddIntegrationButton extends React.Component {
 
   render() {
     // eslint-disable-next-line no-unused-vars
-    const {provider, onAddIntegration, reinstall, ...buttonProps} = this.props;
-    let buttonText = t(reinstall ? 'Reinstall %s' : 'Add %s', provider.metadata.noun);
+    const {provider, buttonText, reinstall, ...buttonProps} = this.props;
+
+    const label =
+      buttonText || t(reinstall ? 'Reinstall %s' : 'Add %s', provider.metadata.noun);
```

Comment:
We should get the reinstall condition out of here. Unrelated but it's a little too specific for this now I think

Reply:
I kept it here for now, since other parts of the app are still using `AddIntegrationButton` as it was before. Once we update those places to use `AddIntegration` with their own button, we can probably remove it from here.

Reply:
Yeah looks good